docs/poly-to-normal-variant/index.md

```diff
@@ -0,0 +1,345 @@
+
+```reason
+discount([> `MissingSandwichTypes
+          | `NeedMegaBurger
+          | `NeedOneBurger
+          | `NeedTwoBurgers ])
+```
+
+This is essentially the same type as before (other than the `>` symbol,
+explained in the next section), just written differently. By looking at the
+usage of the `discount` variable, OCaml can infer how to fill in the type
+variable and produce the fixed type shown above.
+
+The advantage of using a type variable for the definition of the
+`Promo.discount` type is that when you add, rename, or delete variant tags in
+`Discount`, you won't have to make corresponding edits to `Promo.discount`[^1].
+
+## `>` = "allow more than"
+
+In the type expression above, we once again see `>`, so let's see what it means.
+In polymorphic variant type expressions, it means "allow more than". In this
+case, it means that tags other than the four that are listed are allowed. For
+example, this type would be allowed:
+
+```reason{5-6}
+discount([| `MissingSandwichTypes
+          | `NeedMegaBurger
+          | `NeedOneBurger
+          | `NeedTwoBurgers
+          | `HoneyButter
+          | `KewpieMayo ])
+```
+
+When defining your own types, you will most often used *fixed* polymormorphic
+variants, i.e. those that don't have `>` in their type expressions. But it is
+still useful to know what `>` does, since it appears when the compiler
+infers the type of a variable or function that uses polymorphic variants.
+
+::: tip
+
+Fixed polymorphic variants and normal variants are roughly equivalent and can be
+used interchangeably.
+
+:::
+
+## Implicit type variable
+
+Let's come back to the question of why the original attempt at a variant type
+definition was syntactically invalid:
+
+<<< Types.re#bad-discount-type
+
+The reason is that there's an implicit type variable around the `>`. The
+above type expression is equivalent to:
+
+<<< Types.re#explicit-type-var{14}
+
+Now the error message makes a bit more sense:
+
+<<< type-error.txt{7}
+
+The type variable exists, but it's pointless unless it appears as an argument of
+the `discount` type constructor. Once it's added, it compiles:
+
+<<< Types.re#add-type-arg{1}
+
+This is somewhat like accidentally using a variable in a function but forgetting
+to add that variable to the function's argument list.
+
+## Force `DiscountError` argument to be polymorphic variant
+
+Right now the argument of the `DiscountError` constructor can be any type at
+all, but to be explicit, we can force it to be a polymorphic variant:
+
+<<< Types.re#must-be-poly{4}
+
+The `[> ]` type expression means a polymorphic variant that has no tags, but
+allows more tags, which basically means any polymorphic variant. Note that
+adding this small restriction to the type doesn't make a real difference in this
+program---it's just a way to make it clear that `DiscountError`'s argument
+should be a polymorphic variant. It's an optional embellishment that you can
+feel free to leave out.
```

Comment:
If it's constraining the type to poly vars then it's more than an embellishment right? I would remove this last sentence.